### PR TITLE
Fix feature observationType issues

### PR DESCRIPTION
**Pull request by antonoberg**
Depends on https://github.com/DINA-Web/dina-schema/pull/42

### CODE REVIEW

File: src/domainModules/collectionMammals/components/MammalForm/index.spec.js [hunk 1]
```diff
@@ -243,16 +243,26 @@ describe('domainModules/collectionMammals/components/MammalForm', () => {
...
+          'featureObservations[1].featureObservationType.featureObservationTypeName',
+      },
+      {
+        interaction: 'click',
         name:
           'featureObservations[1].featureObservationType.featureObservationTypeName',
         selector: ({ form, name }) => {
-          const first = form.find({ name })
-          return first.find('input').hostNodes()
+          return form
```
Comment (fredrikolovsson) [inline src/domainModules/collectionMammals/components/MammalForm/index.spec.js]: it's not clear from this code that it intends to select the `age` featureObservationTypeName. maybe passing a named function (could be inline) to the selector would solve that.

File: src/domainModules/collectionMammals/components/MammalForm/index.spec.js [hunk 2]
```diff
@@ -243,16 +243,26 @@ describe('domainModules/collectionMammals/components/MammalForm', () => {
...
+            .find({ name })
+            .find('.item')
+            .at(0)
+            .hostNodes()
```
Comment (fredrikolovsson) [inline src/domainModules/collectionMammals/components/MammalForm/index.spec.js:264]: it's not clear from this code that it intends to select the age featureObservationTypeName. maybe passing a named function (could be inline) to the selector would solve that.

File: src/domainModules/collectionMammals/components/MammalForm/transformations/output.js [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+    }
+    case 'ageStage': {
+      return 2
+    }
+    case 'conditionAtCollecting': {
+      return 3
+    }
+    case 'sex': {
+      return 4
+    }
```
Comment (fredrikolovsson) [inline src/domainModules/collectionMammals/components/MammalForm/transformations/output.js]: these IDs don't match the DB. the DB seems to have more types than the frontend and different IDs. check with Ida
screenshot 2018-01-21 18 35 42

**Review by fredrikolovsson — CHANGES_REQUESTED** [review]

File: src/domainModules/collectionMammals/components/MammalForm/index.spec.js (hunk 2 above)
Comment (antonoberg) [reply src/domainModules/collectionMammals/components/MammalForm/index.spec.js:264]: Correct. Will instead use a selector that selects the specific field instead

File: src/domainModules/collectionMammals/components/MammalForm/transformations/output.js (hunk 3 above)
Comment (antonoberg) [reply src/domainModules/collectionMammals/components/MammalForm/transformations/output.js]: I have sent the question to Ida

File: src/domainModules/collectionMammals/components/MammalForm/index.spec.js (hunk 2 above)
Comment (fredrikolovsson) [reply src/domainModules/collectionMammals/components/MammalForm/index.spec.js:264]: ok as discussed we'll fix this later

File: src/domainModules/collectionMammals/components/MammalForm/transformations/output.js [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+  switch (featureObservationTypeName) {
+    case 'sex': {
+      return 1
+    }
+    case 'length': {
+      return 2
+    }
+    case 'age': {
+      return 3
+    }
```
Comment (fredrikolovsson) [inline src/domainModules/collectionMammals/components/MammalForm/transformations/output.js:13]: add case `weight` with id 4

**Review by fredrikolovsson — COMMENTED** [review]
maybe we should revisit the input transformation that starts in line 24 that checks for a conditionAtCollecting observation. that transformation might be outdated now (at least it feels confusing), but we could also check that with the team later

I merged the schema, please add the missing case and update dependencies and it should be good to go

**Pull request merged**